### PR TITLE
Fix: Permission denied accessing error message

### PR DESCRIPTION
**Pull request by MoumitaM**
## Description of the change

- Added error handling for errors of type string 
- Moved the error message accessing line under try-catch block

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [x] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

- - -
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/613)

### CODE REVIEW

File: analytics.js [hunk 1]
```diff
@@ -853,7 +853,14 @@ class Analytics {
             }
           }
         } catch (err) {
-          err.message = `[sendToNative]:: [Destination: ${obj.name}]:: "${err.message}"`;
+          if (err instanceof Error) {
+            err.message = `[sendToNative]:: [Destination: ${obj.name}]:: "${err.message}"`;
+          } else if (typeof err === 'string') {
```
Comment (saikumarrs) [inline analytics.js:858]: Are there any places in the SDK where the errors are thrown as strings?

File: analytics.js [hunk 2]
```diff
@@ -853,7 +853,14 @@ class Analytics {
...
+            // eslint-disable-next-line no-ex-assign
+            err = {
```
Comment (saikumarrs) [inline analytics.js:860]: Don't re-assign. Instead, declare a new variable.

File: utils/utils.js [hunk 3]
```diff
@@ -128,8 +128,8 @@ function notifyError(error) {
 }
 
 function handleError(error, analyticsInstance) {
-  let errorMessage = error.message;
   try {
+    let errorMessage = error.message;
```
Comment (saikumarrs) [inline utils/utils.js:132]: While this fix solves the problem, we'll lose out on the original error. Add error handling around this statement and default to an empty message on error.

File: analytics.js [hunk 4]
```diff
@@ -853,7 +853,14 @@ class Analytics {
             }
           }
         } catch (err) {
-          err.message = `[sendToNative]:: [Destination: ${obj.name}]:: "${err.message}"`;
+          if (err instanceof Error) {
+            err.message = `[sendToNative]:: [Destination: ${obj.name}]:: "${err.message}"`;
```
Comment (saikumarrs) [inline analytics.js:857]: Can we avoid repetition of this message format?

**Review by saikumarrs — CHANGES_REQUESTED** [review]

File: analytics.js (hunk 1 above)
Comment (MoumitaM) [reply analytics.js:858]: Not from core SDK but from integration SDKs

File: analytics.js (hunk 2 above)
Comment (MoumitaM) [reply analytics.js]: Addressed.

File: utils/utils.js (hunk 3 above)
Comment (MoumitaM) [reply utils/utils.js]: Can you check the latest implementation and confirm this change is still needed?

File: analytics.js (hunk 4 above)
Comment (MoumitaM) [reply analytics.js]: Addressed.

File: analytics.js (hunk 1 above)
Comment (saikumarrs) [reply analytics.js:861]: Can you fix those?

File: utils/utils.js (hunk 3 above)
Comment (saikumarrs) [reply utils/utils.js]: No. It still doesn't address the "permission denied" issue.

File: analytics.js (hunk 1 above)
Comment (MoumitaM) [reply analytics.js:861]: At this moment it is not possible.

File: utils/utils.js (hunk 3 above)
Comment (MoumitaM) [reply utils/utils.js]: Addressed

**Review by saikumarrs — APPROVED** [review]

**Pull request merged**